staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -462,9 +459,6 @@ func withRetries(maxRetries int, f func() ([]*metav1.APIGroup, []*metav1.APIReso
 func setDiscoveryDefaults(config *restclient.Config) error {
 	config.APIPath = ""
 	config.GroupVersion = nil
-	if config.Timeout == 0 {
```

Comment:
We shouldn't remove the default timeout to preserve backwards compatibility.

What if we add -1 for no timeout instead?

Reply:
Actually I think this is already the current behavior?

```
kubectl get nodes --request-timeout=-1s -v10

...

I0719 14:20:47.145520   41053 round_trippers.go:435] curl ... foo.com/api/v1/nodes?limit=500'
```

Reply:
it looks like it's an issue specific to the kubectl documentation. Unless there is a mismatch between the client-go documentation and client-go behavior, we shouldn't change client-go.

Reply:
I believe this is a doc-code dismatch. Other places have the correct behavior while only this one misbehaves.

Reply:
`kubectl options` shows the following.
```text
      --request-timeout='0': The length of time to wait before giving up on a single server request.
Non-zero values should contain a corresponding time unit (e.g. 1s, 2m, 3h). A value of zero means
don't timeout requests.
```

Note that I did not really check if other places work as documented. Only that when searching the codebase for this particular flag I did not find any other code using it wrongly.

Reply:
+1 to @roycaihw's comment. If we want to respect the --request-timeout documentation in kubectl (which seems reasonable) we should do it by passing in a value to client-go to tell it to not timeout. We cannot change the default in client-go (which is used by more clients than just kubectl), since that would be a breaking change to the other clients.

Reply:
I'm not sure removing timeout entirely makes sense... the server will still time out eventually, regardless of what the client does (and [at around 30 seconds for short-lived requests](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go#L57), I think, at least for REST API requests)